src/components/flowchart-wrapper/flowchart-wrapper.js

```diff
@@ -35,63 +35,158 @@ const linkToFlowchartInitialVal = {
   showGoBackBtn: false,
 };
 
+const getKeyByValue = (object, value) => {
+  return Object.keys(object).find((key) => object[key] === value);
+};
+
 /**
  * Main flowchart container. Handles showing/hiding the sidebar nav for flowchart view,
  * the rendering of the flowchart, as well as the display of all related modals.
  */
 export const FlowChartWrapper = ({
-  flags,
   fullNodeNames,
+  graph,
   loading,
+  metadataVisible,
   modularPipelinesTree,
   nodes,
-  onLoadNodeData,
   onToggleFocusMode,
   onToggleModularPipelineActive,
   onToggleModularPipelineExpanded,
-  onUpdateActivePipeline,
+  onToggleNodeSelected,
   pipelines,
   sidebarVisible,
-  metadataVisible,
 }) => {
   const history = useHistory();
+  const { pathname, search } = useLocation();
+  const searchParams = new URLSearchParams(search);
 
-  // Reload state is to ensure it will call redirectLocation
-  // only when the page is reloaded.
-  const [reload, setReload] = useState(false);
+  const [errorMessage, setErrorMessage] = useState({});
+  const [invalidUrl, setInvalidUrl] = useState(false);
+  const [usedNavigationBtn, setUsedNavigationBtn] = useState(false);
 
-  const [counter, setCounter] = React.useState(60);
+  const [counter, setCounter] = useState(60);
   const [goBackToExperimentTracking, setGoBackToExperimentTracking] =
     useState(false);
 
+  const graphRef = useRef(null);
+
+  const {
+    matchedFlowchartMainPage,
+    matchedSelectedPipeline,
+    matchedSelectedNodeId,
+    matchedSelectedNodeName,
+    matchedFocusedNode,
+  } = findMatchedPath(pathname, search);
+
+  const resetErrorMessage = () => {
+    setErrorMessage({});
+    setInvalidUrl(false);
+  };
+
+  const redirectToSelectedNode = () => {
+    const node =
+      searchParams.get(params.selected) ||
+      searchParams.get(params.selectedName);
+
+    const nodeId =
+      getKeyByValue(fullNodeNames, node) ||
+      Object.keys(nodes).find((nodeId) => nodeId === node);
+
+    if (nodeId) {
+      const modularPipeline = nodes[nodeId];
+      const hasModularPipeline = modularPipeline?.length > 0;
+
+      if (hasModularPipeline) {
+        onToggleModularPipelineExpanded(modularPipeline);
+      }
+      onToggleNodeSelected(nodeId);
+      resetErrorMessage();
```

Comment:
should we resetErrorMessage everytime ? Can we call resetErrorMessage function only when there is already an error message displayed ?